packages/cli/src/models/local/LocalBaseController.js

```diff
@@ -110,6 +115,57 @@ export default class LocalBaseController {
     return this.hasTypeIdentifier(contractDataPath, "t_function_baredelegatecall_nonpayable$__$returns$_t_bool_$")
   }
 
+  getUninitializedBaseContracts(contractDataPath) {
+    if (!fs.exists(contractDataPath)) return []
+    const contractJson = fs.parseJson(contractDataPath)
+    // Check whether the contract has base contracts
+    const baseContracts = contractJson.ast.nodes.find(n => n.name === contractJson.contractName).baseContracts
+    if (baseContracts.length == 0) {
+      return []
+    }
+    // Make a dict of base contracts that have "initialize" function
+    const baseContractsWithInitialize = {}
+    for (const baseContract of baseContracts) {
+      const baseContractName = baseContract.baseName.name
+      const baseContractPath = Contracts.getLocalPath(baseContractName)
+      const baseContractAbi = fs.parseJson(baseContractPath).abi
+      if (baseContractAbi.find(fn => fn.type === "function" && fn.name === "initialize")) {
+        baseContractsWithInitialize[baseContractName] = false
+      }
+    }
+    // Find "initialize" function in the AST
+    const contractDefinition = contractJson.ast.nodes
+      .find(n => n.nodeType === "ContractDefinition" && n.name === contractJson.contractName)
+    const initializeFunction = contractDefinition.nodes
+      .find(n => n.nodeType === "FunctionDefinition" && n.name === "initialize")
+    if (initializeFunction === undefined) {
+      return []
```

Comment:
This is an interesting scenario: what if the child contract has no initialize function, but its parent(s) do?

If the child contract has a single parent, then the user can call that `initialize` directly, as the child will inherit it. But what if it has multiple parents, each with its own `initialize`? In that case, it'd seem like the child contract **must** have an `initialize` to call its parents', otherwise only one of the parents would get initialized.

Using contracts like the ones you defined:
```solidity
contract ShouldHaveInitialize is WithInitialize, AnotherWithInitialize {
}

contract DoesNotNeedAnInitialize is WithInitialize {
}
```

Let me know if you agree, or if I'm missing something!

Reply:
That's right!